### PR TITLE
Add Support for LayerGroups

### PR DESCRIPTION
**Pull request by hblitza**
This adds a class `LayerGroupClient` and initiates it with a `get` and `modify` method.  

@chrismayer please check

### CODE REVIEW

File: src/layergroup.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import fetch from 'node-fetch';
+import { getGeoServerResponseText, GeoServerResponseError } from './util/geoserver.js';
+import AboutClient from './about.js'
+
+/**
+ * Client for GeoServer layergroups
+ *
+ * @module LayerClient
```
Comment (chrismayer) [inline src/layergroup.js:8]:
```suggestion
 * @module LayerGroupClient
```

**Review by chrismayer — CHANGES_REQUESTED** [review]
Thanks for this @hblitza. Looks very good. Please refer to the comments in the review and please also ensure that you add at least a test for each function.

**Comment by hblitza** [comment]
Thx for review!

**Comment by chrismayer** [comment]
Thanks for your ongoing work @hblitza. As stated in my review, please add at least one test for each function before we can merge this.

**Comment by hblitza** [comment]
Right, I will add a `create layergroup` method. while working on the tests, it became clear, we need one for testing...

**Comment by hblitza** [comment]
@chrismayer please check again.

File: src/layergroup.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+   * @property {number} maxx The maximum x coordinates. Default: 180
+   * @property {number} maxy The maximum y coordinates. Default: 90
+   * @property {String} crs The crs of the bounds. Default: 'EPSG:4326'
+   */
+
+  /**
+   * Create a GeoServer layergroup by the given workspace, layerGroupName, layers and options
+   * @param {String} workspace The name of the workspace
+   * @param {String} layerGroupName The name of the layer group
+   * @param {String} layerGroupTitle The title of the layer group
```
Comment (chrismayer) [inline src/layergroup.js:36]: This seems to be handled by  `layerGroupOptions`, so it can be removed here.

File: src/layergroup.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+      } else {
+        // There was a general problem with GeoServer
+        const geoServerResponse = await getGeoServerResponseText(response);
+        throw new GeoServerResponseError(null, geoServerResponse);
+      }
+    }
+    // return resource location in header
+    const location = response.headers.get('location');
+
+    return location;
```
Comment (chrismayer) [inline src/layergroup.js:112]: Could you give brief info, what exactly is returned here?

Maybe we should harmonize the return values especially for the create functions of the different clients in general. But unrelated to this PR.

**Review by chrismayer — APPROVED** [review]
Thanks for your ongoing work @hblitza. Looks good to me now. 2 minor comments you might want to have a look at before merging.

File: src/layergroup.js (hunk 3 above)
Comment (chrismayer) [reply src/layergroup.js:112]: See https://github.com/meggsimum/geoserver-node-client/issues/143

**Comment by hblitza** [comment]
Right, thanks!

File: src/layergroup.js (hunk 3 above)
Comment (hblitza) [reply src/layergroup.js:112]: good point!

**Comment by chrismayer** [comment]
Thanks for your ongoing work @hblitza. I am going to merge now.

**Pull request merged**